### DISCUSSION TITLE
M5Stack Atom Lite

### DISCUSSION THREAD

**Discussion by Rabbit1543** [revk/xESP32-Faikout#124] [General]
Excuse my ignorance as im not a developer and more a tinkerer. I am trying to install this on a M5stack atom lite using make pico but the flash is failing with the following error. I know its not the device as im able to flash other firmware and the same error happens on other atoms that i have.

"A fatal error occurred: Unable to verify flash chip connection (No serial data received.).
CMake Error at run_serial_tool.cmake:66 (message):"

I have tried using make wroom and i was able to flash that onto another ESP32 board designed for WLED and that worked without issue.

I also tried an s3 build for an M5Stack Stamp S3 which flashed but had problems booting.

**Comment by revk** [comment]
I don't know enough about the M5.

The make just runs the idf.py command to flash, etc.

You can try them directly after making, with whatever parameters the M5 needs.

The options for pico, wroom, or even solo, should cover most variants of the ESP32.

**Reply by Rabbit1543** [reply]
I'll try the other make options and see if I can get them to flash on the atom. It is an esp32-pico-d4 if it means anything. Do you know how to find the parameters you mention and is that done with idf py flash?

**Reply by revk** [reply]
You can manually set anything with make menuconfig

The make options just do a load of presets for that.

I think idf.py has a help option.

**Reply by Rabbit1543** [reply]
Was able to get the s3 build and flash to work if I disabled psram. 

https://shop.m5stack.com/products/m5stamp-esp32s3-module

Was able to get a build for the atom lite to complete but it would not boot. Had to disable psram and lower the minimum chip support to rev1. Also changed the flash size to 4mb and used the 4m.csv for partitions. Not really sure what else to change if anything. Might tinker a bit more but for now the S3 should work for me

**Reply by revk** [reply]
Yeh, basically not a chip config that the Makefile is designed for. So probably possible. But you are a bit on your own trying things out.
